# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1483708138 18000
#      Fri Jan 06 08:08:58 2017 -0500
# Node ID 8e9dd0d6e337c31e75dd143d0d245842e6265f6f
# Parent  71f19f38cbadcb3a7f789c557ca8165d60110528
update check for pcre library to use pkg-config (bug #49993)

* configure.ac: Use PKG_CONFIG to check for pcre library.  Define
PCRE_LDFLAGS separately from PCRE_LIBS.  Display PCRE_LDFLAGS in
summary message.
* subst-config-vals.in.sh, subst-cross-config-vals.in.sh: Also
substitute PCRE_LDFLAGS.
* libinterp/corefcn/toplev.cc (F__octave_config_info__): Also include
PCRE_LDFLAGS in the struct.
* libinterp/build-env.in.cc, libinterp/build-env.h (PCRE_LDFLAGS):
New variable.

diff --git a/build-aux/subst-config-vals.in.sh b/build-aux/subst-config-vals.in.sh
--- a/build-aux/subst-config-vals.in.sh
+++ b/build-aux/subst-config-vals.in.sh
@@ -130,16 +130,17 @@ octincludedir="@octincludedir@"
 octlibdir="@octlibdir@"
 OCT_LINK_DEPS="@OCT_LINK_DEPS@"
 OCT_LINK_OPTS="@OCT_LINK_OPTS@"
 OPENGL_LIBS="@OPENGL_LIBS@"
 OSMESA_CPPFLAGS="@OSMESA_CPPFLAGS@"
 OSMESA_LDFLAGS="@OSMESA_LDFLAGS@"
 OSMESA_LIBS="@OSMESA_LIBS@"
 PCRE_CPPFLAGS="@PCRE_CPPFLAGS@"
+PCRE_LDFLAGS="@PCRE_LDFLAGS@"
 PCRE_LIBS="@PCRE_LIBS@"
 prefix="@prefix@"
 PTHREAD_CFLAGS="@PTHREAD_CFLAGS@"
 PTHREAD_LIBS="@PTHREAD_LIBS@"
 QHULL_CPPFLAGS="@QHULL_CPPFLAGS@"
 QHULL_LDFLAGS="@QHULL_LDFLAGS@"
 QHULL_LIBS="@QHULL_LIBS@"
 QRUPDATE_CPPFLAGS="@QRUPDATE_CPPFLAGS@"
@@ -293,16 +294,17 @@ config_opts="@config_opts@"
   -e "s|%OCTAVE_CONF_OCTLIBDIR%|\"${octlibdir}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_DEPS%|\"${OCT_LINK_DEPS}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_OPTS%|\"${OCT_LINK_OPTS}\"|" \
   -e "s|%OCTAVE_CONF_OPENGL_LIBS%|\"${OPENGL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_CPPFLAGS%|\"${OSMESA_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_LDFLAGS%|\"${OSMESA_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_LIBS%|\"${OSMESA_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_CPPFLAGS%|\"${PCRE_CPPFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_PCRE_LDFLAGS%|\"${PCRE_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_LIBS%|\"${PCRE_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PREFIX%|\"${prefix}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_CFLAGS%|\"${PTHREAD_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_LIBS%|\"${PTHREAD_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_CPPFLAGS%|\"${QHULL_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LDFLAGS%|\"${QHULL_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LIBS%|\"${QHULL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QRUPDATE_CPPFLAGS%|\"${QRUPDATE_CPPFLAGS}\"|" \
diff --git a/build-aux/subst-cross-config-vals.in.sh b/build-aux/subst-cross-config-vals.in.sh
--- a/build-aux/subst-cross-config-vals.in.sh
+++ b/build-aux/subst-cross-config-vals.in.sh
@@ -134,16 +134,17 @@ octincludedir="@octincludedir@"
 octlibdir="@octlibdir@"
 OCT_LINK_DEPS="@OCT_LINK_DEPS@"
 OCT_LINK_OPTS="@OCT_LINK_OPTS@"
 OPENGL_LIBS="@OPENGL_LIBS@"
 OSMESA_CPPFLAGS="@OSMESA_CPPFLAGS@"
 OSMESA_LDFLAGS="@OSMESA_LDFLAGS@"
 OSMESA_LIBS="@OSMESA_LIBS@"
 PCRE_CPPFLAGS="@PCRE_CPPFLAGS@"
+PCRE_LDFLAGS="@PCRE_LDFLAGS@"
 PCRE_LIBS="@PCRE_LIBS@"
 prefix="@prefix@"
 PTHREAD_CFLAGS="@PTHREAD_CFLAGS@"
 PTHREAD_LIBS="@PTHREAD_LIBS@"
 QHULL_CPPFLAGS="@QHULL_CPPFLAGS@"
 QHULL_LDFLAGS="@QHULL_LDFLAGS@"
 QHULL_LIBS="@QHULL_LIBS@"
 QRUPDATE_CPPFLAGS="@QRUPDATE_CPPFLAGS@"
@@ -297,16 +298,17 @@ config_opts="@config_opts@"
   -e "s|%OCTAVE_CONF_OCTLIBDIR%|\"${octlibdir}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_DEPS%|\"${OCT_LINK_DEPS}\"|" \
   -e "s|%OCTAVE_CONF_OCT_LINK_OPTS%|\"${OCT_LINK_OPTS}\"|" \
   -e "s|%OCTAVE_CONF_OPENGL_LIBS%|\"${OPENGL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_CPPFLAGS%|\"${OSMESA_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_LDFLAGS%|\"${OSMESA_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_OSMESA_LIBS%|\"${OSMESA_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_CPPFLAGS%|\"${PCRE_CPPFLAGS}\"|" \
+  -e "s|%OCTAVE_CONF_PCRE_LDFLAGS%|\"${PCRE_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PCRE_LIBS%|\"${PCRE_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_PREFIX%|\"${prefix}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_CFLAGS%|\"${PTHREAD_CFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_PTHREAD_LIBS%|\"${PTHREAD_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_CPPFLAGS%|\"${QHULL_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LDFLAGS%|\"${QHULL_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_QHULL_LIBS%|\"${QHULL_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_QRUPDATE_CPPFLAGS%|\"${QRUPDATE_CPPFLAGS}\"|" \
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1334,70 +1334,37 @@ OCTAVE_CHECK_LIB(qhull, QHull,
   [warn_qhull=
   OCTAVE_CHECK_QHULL_VERSION
   OCTAVE_CHECK_LIB_QHULL_OK(
     [AC_DEFINE(HAVE_QHULL, 1, [Define to 1 if Qhull is available.])],
     [warn_qhull="Qhull library found, but does not seem to work properly.  This will result in loss of functionality of some geometry functions.  Please try recompiling the library with -fno-strict-aliasing."])])
 
 ### Check for PCRE regex library.
 
-PCRE_LIBS=
-
-pcre_fail_msg="to build Octave, you must have the PCRE library and header files installed"
-
-AC_CHECK_HEADERS([pcre.h pcre/pcre.h])
-
-AC_CACHE_CHECK([whether pcre.h defines the macros we need],
-  [ac_cv_pcre_h_macros_present],
-  [AC_EGREP_CPP([PCRE_HAS_MACROS_WE_NEED], [
-    #if defined (HAVE_PCRE_H)
-    # include <pcre.h>
-    #elif defined (HAVE_PCRE_PCRE_H)
-    # include <pcre/pcre.h>
-    #error "NO PCRE HEADER"
-    #endif
-    #if defined (PCRE_INFO_NAMECOUNT) \
-      && defined (PCRE_INFO_NAMEENTRYSIZE) \
-      && defined (PCRE_INFO_NAMETABLE)
-      PCRE_HAS_MACROS_WE_NEED
-    #endif],
-    ac_cv_pcre_h_macros_present=yes, ac_cv_pcre_h_macros_present=no)])
-
-if test $ac_cv_pcre_h_macros_present = yes; then
-  ## check for pcre-config, and if so, get build variables
-  AC_CHECK_PROG(HAVE_PCRE_CONFIG, pcre-config, [yes], [no])
-  if test $HAVE_PCRE_CONFIG = yes; then
-    PCRE_CPPFLAGS=`pcre-config --cflags`
-    PCRE_LIBS=`pcre-config --libs`
-  else
-    PCRE_LIBS="-lpcre"
-  fi
-  save_LIBS="$LIBS"
-  LIBS="$PCRE_LIBS $LIBS"
-  AC_CHECK_FUNCS([pcre_compile],
-    [AC_SUBST(PCRE_CPPFLAGS)
-     AC_SUBST(PCRE_LIBS)],
-    [AC_MSG_ERROR([$pcre_fail_msg])])
-  LIBS="$save_LIBS"
-else
-  AC_MSG_ERROR([$pcre_fail_msg])
-fi
+PKG_CHECK_EXISTS([libpcre],
+  [AC_CHECK_HEADERS([pcre.h pcre/pcre.h])
+   PCRE_CPPFLAGS=`$PKG_CONFIG --cflags-only-I libpcre`
+   PCRE_LDFLAGS=`$PKG_CONFIG --libs-only-L libpcre`
+   PCRE_LIBS=`$PKG_CONFIG --libs-only-l libpcre`
+   AC_SUBST(PCRE_CPPFLAGS)
+   AC_SUBST(PCRE_LDFLAGS)
+   AC_SUBST(PCRE_LIBS)],
+  [AC_MSG_ERROR([to build Octave, you must have the PCRE library and header files installed])])
 
 ### Check for ZLIB library.
 
 OCTAVE_CHECK_LIB(z, ZLIB,
   [ZLIB library not found.  Octave will not be able to save or load compressed data files or HDF5 files.],
   [zlib.h], [gzclearerr])
 
 ### Also define HAVE_ZLIB if libz is found.
 if test -z "$warn_z"; then
   AC_DEFINE(HAVE_ZLIB, 1, [Define to 1 if ZLIB is available.])
 fi
 
-
 ### Check for BZIP2 library.
 
 OCTAVE_CHECK_LIB(bz2, BZIP2,
   [BZIP2 library not found.  Octave will not be able to compress or decompress bzip2 files.],
   [bzlib.h], [BZ2_bzCompressInit])
 
 
 ### Check for the LLVM library
@@ -3019,17 +2986,17 @@ AC_SUBST(SPARSE_XLDFLAGS)
 AC_SUBST(SPARSE_XLIBS)
 
 GNULIB_LINK_DEPS="$COPYSIGNF_LIBM $COPYSIGN_LIBM $FLOORF_LIBM $FLOOR_LIBM $GETHOSTNAME_LIB $LIBSOCKET $LIB_NANOSLEEP $LIB_SELECT $LTLIBINTL $ROUNDF_LIBM $ROUND_LIBM $TRUNCF_LIBM $TRUNC_LIBM"
 
 AC_SUBST(GNULIB_LINK_DEPS)
 
 LIBOCTAVE_LINK_DEPS="$GNULIB_LINK_DEPS $CURL_LIBS $SPARSE_XLIBS $ARPACK_LIBS $QRUPDATE_LIBS $FFTW_XLIBS $LAPACK_LIBS $BLAS_LIBS $READLINE_LIBS $TERM_LIBS $LIBGLOB $PCRE_LIBS $DL_LIBS $PTHREAD_LIBS $FLIBS $LIBS"
 
-LIBOCTAVE_LINK_OPTS="$CURL_LDFLAGS $SPARSE_XLDFLAGS $ARPACK_LDFLAGS $QRUPDATE_LDFLAGS $FFTW_XLDFLAGS"
+LIBOCTAVE_LINK_OPTS="$CURL_LDFLAGS $SPARSE_XLDFLAGS $ARPACK_LDFLAGS $QRUPDATE_LDFLAGS $FFTW_XLDFLAGS $PCRE_LDFLAGS"
 
 AC_SUBST(LIBOCTAVE_LINK_DEPS)
 AC_SUBST(LIBOCTAVE_LINK_OPTS)
 
 if test $ENABLE_DYNAMIC_LINKING = yes; then
   LIBOCTINTERP_LINK_DEPS=""
 else
   LIBOCTINTERP_LINK_DEPS="$DLDFCN_LIBS"
@@ -3318,16 +3285,17 @@ Octave is now configured for $canonical_
   Magick++ CPPFLAGS:             $MAGICK_CPPFLAGS
   Magick++ LDFLAGS:              $MAGICK_LDFLAGS
   Magick++ libraries:            $MAGICK_LIBS
   OPENGL libraries:              $OPENGL_LIBS
   OSMesa CPPFLAGS:               $OSMESA_CPPFLAGS
   OSMesa LDFLAGS:                $OSMESA_LDFLAGS
   OSMesa libraries:              $OSMESA_LIBS
   PCRE CPPFLAGS:                 $PCRE_CPPFLAGS
+  PCRE LDFLAGS:                  $PCRE_LDFLAGS
   PCRE libraries:                $PCRE_LIBS
   PortAudio CPPFLAGS:            $PORTAUDIO_CPPFLAGS
   PortAudio LDFLAGS:             $PORTAUDIO_LDFLAGS
   PortAudio libraries:           $PORTAUDIO_LIBS
   PTHREAD flags:                 $PTHREAD_CFLAGS
   PTHREAD libraries:             $PTHREAD_LIBS
   QHULL CPPFLAGS:                $QHULL_CPPFLAGS
   QHULL LDFLAGS:                 $QHULL_LDFLAGS
diff --git a/libinterp/build-env.h b/libinterp/build-env.h
--- a/libinterp/build-env.h
+++ b/libinterp/build-env.h
@@ -131,16 +131,17 @@ namespace octave
     extern const char *OCTLIBDIR;
     extern const char *OCT_LINK_DEPS;
     extern const char *OCT_LINK_OPTS;
     extern const char *OPENGL_LIBS;
     extern const char *OSMESA_CPPFLAGS;
     extern const char *OSMESA_LDFLAGS;
     extern const char *OSMESA_LIBS;
     extern const char *PCRE_CPPFLAGS;
+    extern const char *PCRE_LDFLAGS;
     extern const char *PCRE_LIBS;
     extern const char *PREFIX;
     extern const char *PTHREAD_CFLAGS;
     extern const char *PTHREAD_LIBS;
     extern const char *QHULL_CPPFLAGS;
     extern const char *QHULL_LDFLAGS;
     extern const char *QHULL_LIBS;
     extern const char *QRUPDATE_CPPFLAGS;
diff --git a/libinterp/build-env.in.cc b/libinterp/build-env.in.cc
--- a/libinterp/build-env.in.cc
+++ b/libinterp/build-env.in.cc
@@ -232,16 +232,18 @@ namespace octave
     const char *OSMESA_CPPFLAGS = %OCTAVE_CONF_OSMESA_CPPFLAGS%;
 
     const char *OSMESA_LDFLAGS = %OCTAVE_CONF_OSMESA_LDFLAGS%;
 
     const char *OSMESA_LIBS = %OCTAVE_CONF_OSMESA_LIBS%;
 
     const char *PCRE_CPPFLAGS = %OCTAVE_CONF_PCRE_CPPFLAGS%;
 
+    const char *PCRE_LDFLAGS = %OCTAVE_CONF_PCRE_LDFLAGS%;
+
     const char *PCRE_LIBS = %OCTAVE_CONF_PCRE_LIBS%;
 
     const char *PREFIX = %OCTAVE_CONF_PREFIX%;
 
     const char *PTHREAD_CFLAGS = %OCTAVE_CONF_PTHREAD_CFLAGS%;
 
     const char *PTHREAD_LIBS = %OCTAVE_CONF_PTHREAD_LIBS%;
 
diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -518,16 +518,17 @@ specified option.
     { "OCTAVE_LINK_OPTS", octave::build_env::OCTAVE_LINK_OPTS },
     { "OCT_LINK_DEPS", octave::build_env::OCT_LINK_DEPS },
     { "OCT_LINK_OPTS", octave::build_env::OCT_LINK_OPTS },
     { "OPENGL_LIBS", octave::build_env::OPENGL_LIBS },
     { "OSMESA_CPPFLAGS", octave::build_env::OSMESA_CPPFLAGS },
     { "OSMESA_LDFLAGS", octave::build_env::OSMESA_LDFLAGS },
     { "OSMESA_LIBS", octave::build_env::OSMESA_LIBS },
     { "PCRE_CPPFLAGS", octave::build_env::PCRE_CPPFLAGS },
+    { "PCRE_LDFLAGS", octave::build_env::PCRE_LDFLAGS },
     { "PCRE_LIBS", octave::build_env::PCRE_LIBS },
     { "PTHREAD_CFLAGS", octave::build_env::PTHREAD_CFLAGS },
     { "PTHREAD_LIBS", octave::build_env::PTHREAD_LIBS },
     { "QHULL_CPPFLAGS", octave::build_env::QHULL_CPPFLAGS },
     { "QHULL_LDFLAGS", octave::build_env::QHULL_LDFLAGS },
     { "QHULL_LIBS", octave::build_env::QHULL_LIBS },
     { "QRUPDATE_CPPFLAGS", octave::build_env::QRUPDATE_CPPFLAGS },
     { "QRUPDATE_LDFLAGS", octave::build_env::QRUPDATE_LDFLAGS },
